src/modules/content-files/test/filedb.c

```diff
@@ -0,0 +1,128 @@
+/************************************************************\
+ * Copyright 2020 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include "src/common/libtap/tap.h"
+#include "src/modules/content-files/filedb.h"
+#include "src/common/libutil/unlink_recursive.h"
+
+void test_badargs (const char *dbpath)
+{
+    void *data;
+    size_t size;
+    const char *errstr;
+    char longkey[8192];
+
+    memset (longkey, 'x', sizeof (longkey));
+    longkey[sizeof (longkey) - 1] = '\0';
+
+    /* get */
+
+    errno = 0;
+    errstr = NULL;
+    ok (filedb_get (dbpath, "/", &data, &size, &errstr) < 0 && errno == EINVAL,
+        "filedb_get key=\"/\" failed with EINVAL");
+    ok (errstr != NULL,
+        "and error string was set");
+
+    errno = 0;
+    errstr = NULL;
+    ok (filedb_get (dbpath, longkey, &data, &size, &errstr) < 0
+        && errno == EOVERFLOW,
+        "filedb_get key=<long> failed with EOVERFLOW");
+    ok (errstr != NULL,
+        "and error string was set");
+
+    errno = 0;
+    ok (filedb_get (dbpath, "noexist", &data, &size, &errstr) < 0
+        && errno == ENOENT,
+        "filedb_get key="" failed with ENOENT");
+
+    /* put */
+
+    errno = 0;
+    errstr = NULL;
+    ok (filedb_put (dbpath, "", "", 2, &errstr) < 0 && errno == EINVAL,
```

Comment:
although it probably doesn't matter for this and the next test, but size should be 1?